manifests/base/service.yaml

```diff
@@ -8,12 +8,16 @@ metadata:
     prometheus.io/port: "8080"
   labels:
     app: training-operator
-  name: training-operator
+  name: training-operator-metrics-service
```

Comment:
Can this file be created in the `rhoai` overlay?

Reply:
Above you mention the pod monitor that gets created by the overlay and removes this service too.
Should the alerting rule not point to the pod monitor service instead of us changing anything training operator side?

Reply:
Absolutely, that would be ideal. I haven't double check what are the requirements, but if PodMonitors work then we would not need that PR AFAIU.

Reply:
@zdtsw Would pointing the alerting rules to the pod monitor service created by the rhoai overlay be sufficient in relation to  [RHOAIENG-6431](https://issues.redhat.com/browse/RHOAIENG-6431)

Reply:
@Bobbins228 these are two different use cases: 6431 is for the SRE monitoring, here is for end user monitoring.

Reply:
ah, i finally understand what this PR is about.
to add the service which can be used by Operator's SRE monitoring.
if you keep the part in Operator then you will need a service in KFTO

Reply:
but why not just update the one in Operator to use name `training-operator`
then you do not need (most likely) touch changes here

Reply:
Right, my understanding is that a service is required only because the way we've defined the prometheus rule in the first place. I think KServe got it right:

https://github.com/red-hat-data-services/rhods-operator/blob/f7a8e5b35791abfaceaffca93865f00ebcd357c0/config/monitoring/prometheus/apps/prometheus-configs.yaml#L208-L228

It's configured to scrape to Pods directly. @zdtsw is my understanding correct?

@Bobbins228 do you think that's something we can do?

Reply:
correct. @VedantMahabaleshwarkar recently updated it for Kserve to use Pod. 
if you already have PodMonitor, you can follow the same way

Reply:
Great, thanks for the confirmation!

Reply:
Yeah sounds good to me too. I will close this and create a new PR to alter the existing rule.
Thanks Wen and Antonin!

Reply:
@Bobbins228 @KPostOffice it would be good to use that approach for Kueue as well for consistency / simplicity sake.

Reply:
FWIW we changed kserve to get the pod and not the metrics service because the metrics service was broken in upstream (and changing the scrapeconfig was easier). Kserve is the outlier here and almost every other component uses a service for SRE monitoring. 
However, I will admit I have no clue what the pros/cons are of doing it one way or the other 😅